clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1295,21 +1268,21 @@ private void close(Duration timeout, boolean swallowException) {
     void prepareShutdown(final Timer timer, final AtomicReference<Throwable> firstException) {
         if (!groupMetadata.get().isPresent())
             return;
-        maybeAutoCommitSync(autoCommitEnabled, timer);
+
+        if (autoCommitEnabled)
+            maybeAutoCommitSync(timer);
+
         applicationEventHandler.add(new CommitOnCloseEvent());
         completeQuietly(
             () -> {
                 maybeRevokePartitions();
-                applicationEventHandler.addAndGet(new LeaveOnCloseEvent(timer), timer);
+                applicationEventHandler.addAndGet(new LeaveOnCloseEvent(calculateDeadlineMs(timer)));
             },
             "Failed to send leaveGroup heartbeat with a timeout(ms)=" + timer.timeoutMs(), firstException);
     }
 
     // Visible for testing
-    void maybeAutoCommitSync(final boolean shouldAutoCommit,
-                             final Timer timer) {
-        if (!shouldAutoCommit)
-            return;
+    void maybeAutoCommitSync(final Timer timer) {
```

Comment:
This is not a "maybe" anymore, so what about `autoCommitSyncAllConsumed`?

Reply:
Changed to just `autoCommitSync()`. Is that OK?